pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -589,36 +628,6 @@ func (e *Controller) syncService(key string) error {
 	return nil
 }
 
-// checkLeftoverEndpoints lists all currently existing endpoints and adds their
-// service to the queue. This will detect endpoints that exist with no
-// corresponding service; these endpoints need to be deleted. We only need to
-// do this once on startup, because in steady-state these are detected (but
-// some stragglers could have been left behind if the endpoint controller
-// reboots).
```

Comment:
I'm not sure we can safely remove this yet. It's still possible for Services to be deleted while the controller isn't running, and I don't think there's anything in the controller to recover from that. There is the rest code, but that can still fail. I'm not sure if it's really possible now, but it might be good to set an ownerRef on Endpoints that points to the Service - EndpointSlices have done that from the start to avoid having to deal with this kind of cleanup.